cmd/autoscaler/main.go

```diff
@@ -108,22 +107,32 @@ func main() {
 		log.Fatal("Error loading/parsing logging configuration: ", err)
 	}
 	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
-	defer flush(logger)
+	defer logger.Sync()
 	ctx = logging.WithLogger(ctx, logger)
 
+	pprof := k8sruntime.NewProfilingServer(logger.Named("pprof"))
+	mp, tp := otel.SetupObservabilityOrDie(ctx, "autoscaler", logger, pprof)
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		if err := mp.Shutdown(ctx); err != nil {
+			logger.Errorw("Error flushing metrics", zap.Error(err))
+		}
+		if err := tp.Shutdown(ctx); err != nil {
+			logger.Errorw("Error flushing traces", zap.Error(err))
+		}
+	}()
+
 	// statsCh is the main communication channel between the stats server and multiscaler.
 	statsCh := make(chan asmetrics.StatMessage, statsBufferLen)
 	defer close(statsCh)
 
-	profilingHandler := profiling.NewHandler(logger, false)
-
 	cmw := configmap.NewInformedWatcher(kubeclient.Get(ctx), system.Namespace())
 	// Watch the logging config map and dynamically update logging levels.
 	cmw.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
 	// Watch the observability config map
-	cmw.Watch(metrics.ConfigMapName(),
-		metrics.ConfigMapWatcher(ctx, component, nil /* SecretFetcher */, logger),
-		profilingHandler.UpdateFromConfigMap)
+	cmw.Watch(o11yconfigmap.Name(), pprof.UpdateFromConfigMap)
```

Comment:
I realize this applies to a lot of other components (not just the autoscaler), but is there a reason we don't watch any of the other observability config for updates? What should users do if they want to change the metrics protocol or endpoint?

Reply:
> I realize this applies to a lot of other components (not just the autoscaler), but is there a reason we don't watch any of the other observability config for updates? 

pprof is easy enough to update (just setting a bool). For other things (metrics,tracing) In the OTel doc we explicitly called out that we would not support a hot-reload of the configuration to keep things simple.

There was a lot of code in the OpenCensus implementation to support this and it probably wasn't worth the effort.